hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/CopyFilter.java

```diff
@@ -47,13 +53,29 @@ public void initialize() {}
    * @return An instance of the appropriate CopyFilter
    */
   public static CopyFilter getCopyFilter(Configuration conf) {
+    String filtersClassName = conf.get(DistCpConstants.CONF_LABEL_FILTERS_CLASS);
+
+    if (filtersClassName != null) {
+      try {
+        Class<? extends CopyFilter> filtersClass = conf.getClassByName(filtersClassName).asSubclass(CopyFilter.class);
+        filtersClassName = filtersClass.getName();
+        Constructor<? extends CopyFilter> constructor = filtersClass.getDeclaredConstructor(Configuration.class);
+        return constructor.newInstance(conf);
+      } catch (Exception e) {
+        LOG.error("Unable to instantiate " + filtersClassName, e);
+      }
+    }
+    return getDefaultCopyFilter(conf);
```

Comment:
after fixing the class load to propagate the exception, make this an `else {}` clause